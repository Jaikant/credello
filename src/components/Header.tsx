import * as React from 'react';
import styled from 'styled-components';
import { device } from 'libs/device';
import { TextBold, Text } from 'components/Typography';
import ManageDebt from 'components/ManageDebt';

type Props = {
  title?: string;
  subtitle?: string;
};

const HeroWrapper = styled.div`
    background-image: url('${require('../assets/images/Hero.png')}');
    background-size: contain;
    background-repeat: no-repeat;
    background-position: top right;
    margin: -32px -16px;
    padding: 32px 16px;
    @media ${device.tablet} { 
      display: grid;
      grid-gap: 32px;  
      background-image: url('${require('../assets/images/HeroDesktop.png')}');
      background-size: 50% 100%, contain;
      margin: -32px -24px;
      padding: 32px 24px;
    };
    @media ${device.laptop} { 
      display: grid;
      grid-gap: 64px;
      background-image: url('${require('../assets/images/HeroDesktop.png')}');
      background-size: 50% auto, contain;
      margin: -32px -128px;
      padding: 32px 128px;
    }  
`;

const Header = ({ title, subtitle }: Props) => {
  return (
    <HeroWrapper>
      <div>
        <TextBold tag="h1" size="32px" mb="8px" color="#02254d">
          {title}
        </TextBold>
        <Text laptopSize="16px" mb="32px" color="#6c7f87">
          {subtitle}
        </Text>
      </div>
      <ManageDebt />
    </HeroWrapper>
  );
};

Header.defaultProps = {
  title: 'Make financial decisions simple',
  subtitle: 'with personalized guidance',
};

export default Header;
