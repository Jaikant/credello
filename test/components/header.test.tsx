import React from 'react';
import { render } from '../testUtils';
import Header from '../../src/components/Header';
import 'jest-styled-components';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Header />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
