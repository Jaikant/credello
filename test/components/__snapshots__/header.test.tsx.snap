// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  font-family: Roboto;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.2;
  webkit-font-smoothing: antialiased;
  margin: 0;
  margin-bottom: 32px;
  margin-top: 0;
  color: #6c7f87;
  text-align: initial;
}

.c1 {
  font-family: Roboto;
  font-weight: 700;
  font-size: 32px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  color: #02254d;
  text-align: initial;
}

.c3 {
  font-family: Roboto;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  color: #02254d;
  text-align: initial;
}

.c4 {
  display: grid;
  grid-gap: 8px;
  grid-template-rows: 1fr 1fr;
  grid-template-columns: 1fr 1fr;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5cdb95;
  background-image: url('test-file-stub');
  background-size: 100% 100%,cover;
  background-repeat: no-repeat;
  background-position: center;
  text-transform: uppercase;
  border-radius: 5px;
  color: #fff;
  font-weight: 800;
  font-size: 14px;
  line-height: 1.2;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c5 div {
  display: grid;
  text-align: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 16px;
}

.c5:before {
  content: '';
  display: block;
  height: 0px;
  width: 0px;
  background: red;
  padding-bottom: calc(188 / 328 * 100%);
}

.c7 {
  font-family: Roboto;
  font-weight: 500;
  font-size: 12px;
  line-height: 1.2;
  margin: 0;
  margin-bottom: 0;
  margin-top: 8px;
  color: #5cdb95;
  text-align: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  margin: 8px;
}

.c0 {
  background-image: url('test-file-stub');
  background-size: contain;
  background-repeat: no-repeat;
  background-position: top right;
  margin: -32px -16px;
  padding: 32px 16px;
}

@media (min-width:768px) {

}

@media (min-width:1224px) {
  .c2 {
    font-size: 16px;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 32px;
  }
}

@media (min-width:1224px) {
  .c1 {
    line-height: 1.5;
    font-size: 32px;
  }
}

@media (min-width:768px) {

}

@media (min-width:1224px) {
  .c3 {
    line-height: 1.5;
  }
}

@media (min-width:768px) {
  .c4 {
    grid-gap: 32px;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
  }
}

@media (min-width:1224px) {
  .c4 {
    grid-gap: 32px;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
  }
}

@media (min-width:768px) {
  .c7 {
    font-size: 12px;
  }
}

@media (min-width:1224px) {
  .c7 {
    font-size: 12px;
  }
}

@media (min-width:768px) {
  .c6 {
    margin: 16px;
  }
}

@media (min-width:1224px) {
  .c6 {
    margin: 32px;
  }
}

@media (min-width:768px) {
  .c0 {
    display: grid;
    grid-gap: 32px;
    background-image: url('test-file-stub');
    background-size: 50% 100%,contain;
    margin: -32px -24px;
    padding: 32px 24px;
  }
}

@media (min-width:1224px) {
  .c0 {
    display: grid;
    grid-gap: 64px;
    background-image: url('test-file-stub');
    background-size: 50% auto,contain;
    margin: -32px -128px;
    padding: 32px 128px;
  }
}

<div
    class="c0"
  >
    <div>
      <div
        class="c1"
        color="#02254d"
      >
        Make financial decisions simple
      </div>
      <div
        class="c2"
        color="#6c7f87"
      >
        with personalized guidance
      </div>
    </div>
    <div>
      <div
        class="c3"
        color="#02254d"
      >
        Manage your debt better
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
          src="payoff-debt.png"
        >
          <div>
            Ways to pay off debt
          </div>
        </div>
        <div
          class="c5"
          src="estimate-debt.png"
        >
          <div>
            Estimate debt free date
          </div>
        </div>
        <div
          class="c5"
          src="consolidate.png"
        >
          <div>
            Ways to consolidate
          </div>
        </div>
        <div
          class="c5"
          src="estimate-savings.png"
        >
          <div>
            Estimate savings
          </div>
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="Typography__TextMedium-sc-12f2dd0-1 c7"
          color="#5cdb95"
          href="/LearnAboutDebt"
        >
           Learn more about debt consolidation 
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
